app/src/main/java/com/example/atheneum/viewmodels/FirebaseRefUtils/DatabaseWriteHelper.java

```diff
@@ -221,43 +222,65 @@ public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseR
-        bookRequestRef.addChildEventListener(new ChildEventListener() {
-            @Override
-            public void onChildAdded(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {
-                final String requesterID = dataSnapshot.getKey();
-                Log.i(TAG, "requester being declined from book acceptance: " + requesterID);
-                notification.setNotificationReceiverID(requesterID);
-                notification.setRequesterID(requesterID);
-                Log.i(TAG, "notification requester ID: " + notification.getRequesterID());
-                declineRequest(requesterID, bookID, notification, false);
-            }
-
-            @Override
-            public void onChildChanged(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {
-
-            }
-
-            @Override
-            public void onChildRemoved(@NonNull DataSnapshot dataSnapshot) {
-
-            }
-
+        bookRequestRef.addListenerForSingleValueEvent(new ValueEventListener() {
             @Override
-            public void onChildMoved(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {
-
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                if (dataSnapshot.exists()) {
+                    HashMap<String, Object> updates = new HashMap<>();
+                    for (DataSnapshot data : dataSnapshot.getChildren()) {
+                        String requesterID = data.getKey();
+                        Log.i(TAG, "requester being declined from book acceptance: " + requesterID);
+                        notification.setNotificationReceiverID(requesterID);
+                        notification.setRequesterID(requesterID);
+                        Log.i(TAG, "notification requester ID: " + notification.getRequesterID());
+                        addDeclinedRequestToMultiPathUpdate(requesterID, bookID, notification, updates);
+                    }
+                    // After adding all the declined requests, send the updates
+                    RootRefUtils.ROOT_REF.updateChildren(updates, new DatabaseReference.CompletionListener() {
+                        @Override
+                        public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseReference databaseReference) {
+                            if (databaseError != null) {
+                                Log.e(TAG, "Failed to declineAllRequests!: " + databaseError.toString());
+                            } else {
+                                Log.i(TAG, "Successfully declined all requests!");
+                            }
+                        }
+                    });
+                }
             }
 
             @Override
             public void onCancelled(@NonNull DatabaseError databaseError) {
-
+                Log.e(TAG, databaseError.toString());
             }
         });
     }
 
+    private static void addDeclinedRequestToMultiPathUpdate(String requesterID, String bookID,

```

Comment:
This was not ideal but I didn't want to refactor declineRequest to use this since that would honestly be too dangerous in this stage. I don't want to break the existing code since the code cleanup isn't worth it.